analyses/survival-analysis/util/survival_models.R

```diff
@@ -0,0 +1,117 @@
+# Functions for conducting survival analyses
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2019
+#
+# Install the packages we need
+if (!("survival" %in% installed.packages())) {
+  install.packages("survival")
+}
+
+if (!("survminer" %in% installed.packages())) {
+  install.packages("survminer")
+}
+
+# Attach this package
+library(survminer)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+survival_analysis <- function(metadata, ind_var, test = "kap.meier", ind_data = NULL) {
+  # Given the overall survival information, and an independent variable will
+  # run survival analysis and return a list with 1) the model fit object and
+  # 2) a summary table 3) the data used.
+  #
+  # Args:
+  #   metadata: a data.frame that contains columns OS_status and OS_days to use
+  #             for the survival model. This also assumes "LIVING" and "DECEASED"
+  #             are the two statuses. THis will be converted to a numeric variable
+  #             for use with `survival` R functions. Samples with NAs are dropped.
+  #   ind_var: a character string noting the name of the independent variable to
+  #            test as a predictor for survival.
+  #   test: a character string noting which test to use. Supported choices:
+  #         "kap.meier", "log.rank", "cox.reg"
+  #   ind_data: If the data for the independent variable needed for the test is not
+  #         in metadata, add it here. This assumes it will be a data.frame with
+  #         either a "Tumor_Sample_Barcode", "Kids_First_Biospecimen_ID", or
+  #         "sample" column which to inner_join with metadata by.
+  #
+  # Returns:
+  # A list with three objects: 1) the original model fit object,
+  #                            2) the summary table
+  #                            3) the original data.frame
+
+  # List the tests
+  supported_tests <- c("kap.meier", "log.rank", "cox.reg")
+
+  # Check that it is a supported test
+  if (!(test %in% supported_tests)) {
+    stop(paste0(test, "is not a supported test."))
```

Comment:
Edit this sanity check as - `paste(test, "is not a supported test. Please specify one of the following:", paste(supported_tests, collapse=", "))`